engine/src/execution_graph/logic_controllers/BatchProcessing.h

```diff
@@ -251,9 +262,23 @@ class TableScan : public PhysicalPlan {
 	: PhysicalPlan(), input(loader, schema, context)
 	{}
 	virtual kstatus run() {
-		while( input.wait_for_next() ) {
+		while ( input.wait_for_next() ) {
 			auto batch = input.next();
-			this->add_to_output_cache(std::move(batch));
+
+			// For large files
+			if (batch->sizeInBytes() > 1073741824) {  // Threshold 1GB
```

Comment:
remove this hard coded value and make it configurable somehow

Reply:
Ok